File: projects/m1/011-note-to-frequency/solutions/alessiacaporaso/note-to-frequency.js
/*  Note Frequency
    C4   261.63
    D4   293.66
    E4   329.63
    F4   349.23
    G4   392.00
    A4   440.00
    B4   439.88   */

// reads the name of the note
const noteUser = prompt('Please, enter the note:').toUpperCase();
//divide the letter from the octave
const noteLetter = noteUser.slice(0,1);
const noteOctave = noteUser.slice(1,2);
//calc the frequency
const calcFrequency = Math.pow(2, 4-noteOctave)
// displays the note's frequency
let frequency;
switch(noteLetter){
  case 'C':
    frequency = (261.63/calcFrequency).toFixed(2)
    alert(frequency);
  break;
  case 'D':
    frequency = (293.66/calcFrequency).toFixed(2)
    alert(frequency);
  break;
  case 'E':
    frequency = (329.63/calcFrequency).toFixed(2)
    alert(frequency);
  break;
  case 'F':
    frequency = (349.23/calcFrequency).toFixed(2)
    alert(frequency);
  break;
  case 'G':
    frequency = (392.00/calcFrequency).toFixed(2)
    alert(frequency);
  break;
  case 'A':
    frequency = (440.00/calcFrequency).toFixed(2)
    alert(frequency);
  break;
  case 'B':
    frequency = (439.88/calcFrequency).toFixed(2)
    alert(frequency);
  break;
}

File: projects/m1/011-note-to-frequency/solutions/PiervitoRinaldi/note-to-frequency.py
# NOTES TO FREQUENCY
# Enter the note and the octave of any notes to know its frequency.

b4 = 493.88
a4 = 440.00
g4 = 392.00
f4 = 349.23
e4 = 329.63
d4 = 293.66
c4 = 261.63

note = input('Enter the note: ')
octave = input('Enter the octave: ')
octave = int(octave)


if note == str('C'):
    print(c4 / 2** (4 - octave), 'Hz')
elif note == str('D'):
    print(d4 / 2** (4 - octave), 'Hz')
elif note == str('E'):
    print(d4 / 2** (4 - octave), 'Hz')
elif note == str('F'):
    print(f4 / 2** (4 - octave), 'Hz')
elif note == str('G'):
    print(g4 / 2** (4 - octave), 'Hz')
elif note == str('A'):
    print(a4 / 2** (4 - octave), 'Hz')
elif note == str('B'):
    print(b4 / 2** (4 - octave), 'Hz')






